native/cocos/core/Root.cpp

```diff
@@ -314,6 +314,7 @@ bool Root::setRenderPipeline(pipeline::RenderPipeline *rppl /* = nullptr*/) {
 
         _pipeline = rppl;
         _pipelineRuntime = std::make_unique<RenderPipelineBridge>(rppl);
+        rppl->setPipelineRuntime(_pipelineRuntime.get());
```

Comment:
js does not support multi-inhertiance and RenderPipeline is derived from Asset. So it cannot inherits PipelineRuntime interface. As result, it must keep a PipelineRuntime pointer to interact with other modules that requires PipelineRuntime.